lib/ansible/module_utils/network/common/facts/facts.py

```diff
@@ -0,0 +1,130 @@
+        :param valid_subsets: The valid subsets
+        :rtype: list
+        :returns: The runable subsets
+        """
+        runable_subsets = set()
+        exclude_subsets = set()
+        minimal_gather_subset = frozenset(['default'])
+
+        for subset in subsets:
+            if subset == 'all':
+                runable_subsets.update(valid_subsets)
+                continue
+            if subset == 'min' and minimal_gather_subset:
+                runable_subsets.update(minimal_gather_subset)
+                continue
+            if subset.startswith('!'):
+                subset = subset[1:]
+                if subset == 'min':
+                    exclude_subsets.update(minimal_gather_subset)
+                    continue
+                if subset == 'all':
+                    exclude_subsets.update(
+                        valid_subsets - minimal_gather_subset)
+                    continue
+                exclude = True
+            else:
+                exclude = False
+
+            if subset not in valid_subsets:
+                self._module.fail_json(msg='Bad subset')
+
+            if exclude:
+                exclude_subsets.add(subset)
+            else:
+                runable_subsets.add(subset)
+
+        if not runable_subsets:
+            runable_subsets.update(valid_subsets)
+        runable_subsets.difference_update(exclude_subsets)
+
+        return runable_subsets
+
+    def get_network_resources_facts(self, net_res_choices, facts_resource_obj_map, resource_facts_type=None, data=None):
+        """
+        :param net_res_choices:
+        :param fact_resource_subsets:
+        :param data: previously collected configuration
+        :return:
+        """
+        if net_res_choices:
+            if 'all' in net_res_choices:
+                net_res_choices.remove('all')
+
+        if net_res_choices:
+            if not resource_facts_type:
+                resource_facts_type = self._gather_network_resources
+
+            restorun_subsets = self.gen_runable(resource_facts_type, net_res_choices)
+            if restorun_subsets:
+                self.ansible_facts['gather_network_resources'] = list(restorun_subsets)
+                instances = list()
+                for key in restorun_subsets:
+                    fact_cls_obj = facts_resource_obj_map.get(key)
+                    if fact_cls_obj:
+                        instances.append(fact_cls_obj(self._module))
+                    else:
+                        self._warnings.extend(["network resource fact gathering for '%s' is not supported" % key])
+
+                for inst in instances:
+                    inst.populate_facts(self._connection, self.ansible_facts, data)
+
+    def get_network_legacy_facts(self, fact_legacy_obj_map, legacy_facts_type=None):
+        if not legacy_facts_type:
+            legacy_facts_type = self._gather_subset
+
+        runable_subsets = self.gen_runable(legacy_facts_type, fact_legacy_obj_map.keys())
```

Comment:
The subset type has to be of type `frozenset` as the `minimal_gather_subset` is of type frozenset
`runable_subsets = self.gen_runable(legacy_facts_type, frozenset(fact_legacy_obj_map.keys())`